### ISSUE TITLE
Unstable Melodic builds with multiple cmake macros

### ISSUE THREAD

**Issue by floweisshardt** [floweisshardt/atf#139]
https://github.com/floweisshardt/atf/pull/127 sometimes breaks Melodic builds: Sometimes there is an error on Travis Melodic builds, but Kinetic is always fine:

```
Traceback (most recent call last):
  File "/opt/ros/melodic/share/catkin/cmake/test/run_tests.py", line 41, in <module>
    sys.exit(main())
  File "/opt/ros/melodic/share/catkin/cmake/test/run_tests.py", line 18, in main
    remove_junit_result(args.results)
  File "/opt/ros/melodic/lib/python2.7/dist-packages/catkin/test_results.py", line 46, in remove_junit_result
    os.remove(filename)
OSError: [Errno 2] No such file or directory: '/root/target_ws/build/atf_test/test_results/atf_test/rostest-test_generated_atf_test_generation_config.xml'
make[3]: *** [CMakeFiles/_run_tests_atf_test_rostest_test_generated_atf_test_generation_config.yaml_recording_ts0_c0_r1_e0_s0_0.test] Error 1
make[2]: *** [CMakeFiles/_run_tests_atf_test_rostest_test_generated_atf_test_generation_config.yaml_recording_ts0_c0_r1_e0_s0_0.test.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/run_tests.dir/rule] Error 2
make: *** [run_tests] Error 2
```

full log: https://gist.github.com/floweisshardt/5a1f03a8517ca8e9c9b4ddaae4653af4

**Comment by floweisshardt** [comment]
something similar happend with kinetic too
```
Traceback (most recent call last):
  File "/opt/ros/kinetic/share/catkin/cmake/test/run_tests.py", line 42, in <module>
    sys.exit(main())
  File "/opt/ros/kinetic/share/catkin/cmake/test/run_tests.py", line 19, in main
    remove_junit_result(args.results)
  File "/opt/ros/kinetic/lib/python2.7/dist-packages/catkin/test_results.py", line 45, in remove_junit_result
    os.remove(filename)
OSError: [Errno 2] No such file or directory: '/root/target_ws/build/atf_test_app_user_result/test_results/atf_test_app_user_result/rostest-test_generated_atf_test_generation_config.xml'
make[3]: *** [CMakeFiles/_run_tests_atf_test_app_user_result_rostest_test_generated_atf_test_generation_config.yaml_recording_ts0_c0_r0_e0_s0_2.test] Error 1
make[2]: *** [CMakeFiles/_run_tests_atf_test_app_user_result_rostest_test_generated_atf_test_generation_config.yaml_recording_ts0_c0_r0_e0_s0_2.test.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/run_tests.dir/rule] Error 2
make: *** [run_tests] Error 2
```

**Comment by floweisshardt** [comment]
fixed in #146
